tests/ashley/test_templatetags_custom_tags.py

```diff
@@ -0,0 +1,265 @@
+
+    @staticmethod
+    def _assign_permissions(forum, group, permissions: List[str]):
+        """Grant a list of permissions for a group on a specific forum."""
+        for perm in permissions:
+            assign_perm(perm, group, forum, True)
+
+    @staticmethod
+    def _trim(answer):
+        """Get rid of space and \n to be able to compare the render of template"""
+        return re.sub(r"\s+", "", answer)
+
+    @staticmethod
+    def test_can_tell_if_the_user_is_instructor_of_this_forum():
+        """
+        Given two users. User1 is student of the forum, User2 is instructor.
+        We control that is_user_instructor can detect if a user is an instructor.
+        Then a forum can be part of multiple contexts. If a user is instructor in one context,
+        he is considered admin of this forum in all contexts. We add a new context to the forum
+        where user is instructor and test that user is now considered as instructor
+        """
+        # load template
+        def get_rendered(topic, user):
+            template = Template(
+                "{% load custom_tags %}"
+                + "{% if topic|is_user_instructor:user %}YES{% else %}NO{% endif %}"
+            )
+            context = Context({"topic": topic, "user": user})
+            rendered = template.render(context)
+
+            return rendered
+
+        # Create two LTI Context
+        context1, context2 = LTIContextFactory.create_batch(2)
+
+        # Initialize two users
+        user1, user2 = UserFactory.create_batch(2)
+
+        # Sync user1 groups in context1 with role "student"
+        context1.sync_user_groups(user1, ["student"])
+        # Sync user1 groups in context2 with role "instructor"
+        context2.sync_user_groups(user1, ["instructor"])
+        # Sync user2 groups in context1 with role "instructor"
+        context1.sync_user_groups(user2, ["instructor"])
+
+        # Create forum and add context1
+        forum = ForumFactory(name="Initial forum name")
+        forum.lti_contexts.add(context1)
+
+        # Set up topic
+        topic = TopicFactory(forum=forum, poster=user1, subject="topic création")
+
+        # Chek that user1 is not instructor
+        assert get_rendered(topic, user1) == "NO"
+        # Chek that user2 is instructor
+        assert get_rendered(topic, user2) == "YES"
+
+        # Add forum to context2 where user1 has role "instructor"
+        forum.lti_contexts.add(context2)
+        # Check that user1 is now instructor as well
+        assert get_rendered(topic, user1) == "YES"
+
+    def test_render_block_is_instructor_of_this_forum(self):
+        """
+        check that the instructor icon is present on the different views
+        as expected
+        """
+        created_by = 'By: <a href="{{url}}">{{ username }}</a>'
+        icon = '<i class="icon_writer fas fa-award" aria-hidden="true" title="Instructor"></i>'
+
+        def get_rendered(topic, user, selector=""):
```

Comment:
I would write the expected html directly in the test instead of going through template rendering because this is what you are trying to test. If your template for some reason renders "hey", then you will test that "hey" is also in your response... so the test will pass but the feature is broken.

Reply:
they were no point on using Template the way I did it, I've replaced it with a f" string.